src/commons/s3.rs

```diff
@@ -235,14 +236,13 @@ mod test {
         };
 
         assert_eq!(
-            serde_yaml::to_string(&bucket).unwrap(),
+            yaml::to_explicit_document_string(&bucket).unwrap(),
             "---
 bucketName: test-bucket-name
-connection:
-  inline:
-    host: host
-    port: 8080
-    accessStyle: VirtualHosted
+connection: !inline
```

Comment:
Hm, does this come from serde_yaml? We might need to go back to the old tagging format somehow…

Reply:
https://github.com/dtolnay/serde-yaml/issues/298

Reply:
The proposed solution in `serde_yaml:0.9.4` is to annotate every enum field with `#[serde(with = "serde_yaml::with::singleton_map")]`.

But this solution does not work together with `JsonSchema`, see GREsau/schemars#89.

One workaround is to use the serde parameters `serialize_with` and `deserialize_with`:

```
#[serde(
    default,
    skip_serializing_if = "Option::is_none",
    serialize_with = "serde_yaml::with::singleton_map::serialize",
    deserialize_with = "serde_yaml::with::singleton_map::deserialize"
)]
pub connection: Option<S3ConnectionDef>,
```

This becomes quite verbose.

In which cases do the YAML tags actually pose a problem?

Reply:
`serde_yaml:0.9.9` introduced [singleton_map_recursive](https://docs.rs/serde_yaml/0.9.9/serde_yaml/with/singleton_map_recursive/index.html) which allows to serialize enums as YAML maps containing one entry in which the key identifies the variant name.

This is now used in `stackable_operator::yaml::serialize_to_explicit_document`. Further annotations are not necessary in the operators.